src/Dialog/Dialog.tsx

```diff
@@ -265,6 +359,7 @@ const _Dialog = React.forwardRef<HTMLDivElement, React.PropsWithChildren<DialogP
     width = 'xlarge',
     height = 'auto',
     footerButtons = [],
+    position = {narrow: 'center', regular: 'center'},
```

Comment:
Could we move this default value to an object outside of the component? e.g.

```tsx
const defaultPosition = {
  narrow: 'center',
  regular: 'center',
};

// ...

const _Dialog = React.forwardRef((props) => {
  const { position = defaultPosition } = props
});
```

This way we don't have to think about the object changing every render later on in the component (which thankfully we don't have to think about this with `sx` because of how you approached this 🙌)